test/shellscript.test.js

```diff
@@ -496,5 +496,59 @@ describe("api.js (shell)", () => {
             assert.strictEqual(env.rendition_foobar, "Unicorn");
             assert.strictEqual(env.rendition_crop_x, "Unicorn");
         });
+
+        it("should pass variables larger than 8000 characters as a file", async () => {
```

Comment:
Also test the behavior when the length of parameters is exactly at the limit (aka 8000 chars).

Reply:
I can appreciate why you'd want that but I think it's pretty sufficient, the goal is met by asserting that we cross well over the established 8000 character limit and the biggest variable is the only one which is written to a file.  The test itself creates an exactly 8000 character value and ensures that it is written exactly to a file.  If the test tries to make too many assumptions about how the code calculates the 8000 character threshold and how that's calculated exactly then it makes it more problematic to tune that behavior over time, I think.  Right now the calculation is based on variable name length + variable value length.   If we wanted to be really cautious then maybe adding two more for space delimiters and equal signs would be better?  What about quote marks?  Anyway, the point is I don't want the test to be overly prescriptive, I just want it to assert that at some point a line is crossed and that triggers the new logic.

As a side note, If we wanted to test more granularly then I think a better test is the utility function itself that does this work.  This would at least result in simpler looking tests that don't require actually executing a shell script.

Reply:
Another think I was wondering about... How are we sure the test would actually fail? It could be running on machines with a larger limit... (in theory).

Reply:
The goal of the tests is not to generate the error condition, rather it is to assert the variables are written to files when we're expecting them to after they grow beyond a set size limit.  ARG_MAX and the resulting command buffer limitations really vary a lot by OS, so I have to assume lowest common denominator so we have consistent behavior.  I think 131072 is the right number, so to add some padding for extra characters, path names, etc I upped the limit to 130500.  That means we could have 572 other characters in the whole command that we're not counting before running into that issue again.  I think we can assume that's not an edge case of major concern.